pkg/cli/admin/catalog/build.go

```diff
@@ -44,25 +61,50 @@ type BuildImageOptions struct {
 	FilterOptions   imagemanifest.FilterOptions
 	ParallelOptions imagemanifest.ParallelOptions
 
-	FromFileDir string
-	FileDir     string
+	FromFileDir  string
+	FileDir      string
+	ReleaseImage string
 }
 
-const defaultFromImage = "quay.io/openshift/origin-operator-registry:latest"
-
 func NewBuildImageOptions(streams genericclioptions.IOStreams) *BuildImageOptions {
-	// Adjust default build opts to use an OpenShift base "from" image
-	defaults := appregistry.DefaultAppregistryBuildOptions()
-	defaults.From = defaultFromImage
-
+	// From must be specified, or we must infer it from a release
+	o := appregistry.DefaultAppregistryBuildOptions()
+	o.From = ""
 	return &BuildImageOptions{
-		AppregistryBuildOptions: defaults,
+		AppregistryBuildOptions: o,
 		IOStreams:               streams,
 		ParallelOptions:         imagemanifest.ParallelOptions{MaxPerRegistry: 4},
 	}
 }
 
-func (o *BuildImageOptions) Complete(cmd *cobra.Command, args []string) error {
+func (o *BuildImageOptions) Complete(f kcmdutil.Factory, cmd *cobra.Command, args []string) error {
+	// expect to extract the base image from an active ocp connection or a release image
+	if len(o.From) == 0 {
+		// get an iostream buffer for extracting the image name from the release image
+		streams, _, out, errOut := genericclioptions.NewTestIOStreams()
+		infoOpts := release.NewInfoOptions(streams)
+		infoOpts.ImageFor = "operator-registry"
+		images := []string{}
+		if len(o.ReleaseImage) > 0 {
+			images = append(images, o.ReleaseImage)
+		}
+		if err := infoOpts.Complete(f, nil, images); err != nil {
```

Comment:
Seems like you could just use `LoadReleaseInfo` from `InfoOptions` which would be much nicer than what you're doing here.